### ISSUE TITLE
[REGRESSION]: Selector `text=""` works differ for element with nested elements since 1.9

### ISSUE THREAD

**Issue by frezot** [microsoft/playwright#5583]
**Context:**
- GOOD Playwright Version: 1.8.1
- BAD Playwright Version: 1.9
- Operating System: Linux, Mac

**Code Snippet**

https://github.com/frezot/pw-1.9-bug

**Describe the bug**

Page has element
```html
<div>Illon M.<span>*9025 SOME BANK</span></div>
```

in 1.8.1 for get this div by text
```
await page.waitForSelector('text="Illon M."')
``` 

in 1.9
```sh
    page.waitForSelector: Timeout 1000ms exceeded.
    =========================== logs ===========================
    waiting for selector "text="Illon M."" to be visible
    ============================================================
```

**Comment by mxschmitt** [comment]
I think this works as expected, see [here](https://playwright.dev/docs/selectors/#text-selector):
> `text="Log in"` - text body can be escaped with single or double quotes for full-string case-sensitive match. For example `text="Log"` does not match `<button>Log in</button>` but instead matches `<span>Log</span>`.

So in your case `await page.waitForSelector('text=Illon M.')` would be the solution, since it's a partial match of the `div` element. Removing the quotes also allows partial matches.

**Comment by KirProkopchik** [comment]
@mxschmitt 
My example is <span data-testid="optionText">25–34</span>
Version 1.8.1 
await page.click('"25–34"'); or await page.click('text=25–34');

Version 1.9.0
 waiting for selector "text=25–34"
      selector resolved to visible <span data-testid="optionText" class="styled__OptionTe…>25–34</span>
    attempting click action
      waiting for element to be visible, enabled and stable
        element is not visible - waiting...

**Comment by dgozman** [comment]
Proposed fix in #5603.

**Comment by aslushnikov** [comment]
The fix was merged in v1.9.1 and 1.9.1 is live!
